### PR TITLE
Replace lrbd with ceph-iscsi

### PR DESCRIPTION
**Pull request by ricardoasmarques**
This PR replaces `lrbd` with `ceph-iscsi`.

### CODE REVIEW

File: xml/deployment_iscsi.xml [hunk 1]
```diff
@@ -305,326 +305,159 @@
...
-}
-</screen>
    </sect3>
    <sect3>
     <title>Discovery and Mutual Authentication</title>
     <para>
      <emphasis>Discovery authentication</emphasis> is independent of the
-     previous authentication methods. It requires credentials for browsing.
+     previous authentication methods. It requires credentials for browsing,
+     it's optional, and can be configured by:
```
Comment (tbazant) [inline xml/deployment_iscsi.xml]: it is optional....

File: xml/deployment_iscsi.xml [hunk 2]
```diff
@@ -305,326 +305,159 @@
...
+      letters, '@', '-', '_' or '/'.
+     </para>
+    </note>
     <para>
-     Authentication of type <option>tpg</option>,
-     <option>tpg+identified</option>, <option>acls</option>, and
-     <option>discovery</option> support mutual authentication. Specifying the
-     mutual settings requires that the target authenticates against the
-     initiator.
+     Optionally, you can also specify the <literal>mutual_username</literal> and
```
Comment (tbazant) [inline xml/deployment_iscsi.xml]: same as before

**Comment by ricardoasmarques** [comment]
Comments addressed, thanks for the feedback. I'm now working on the missing sections and let you know when I submit those sections.

**Comment by ricardoasmarques** [comment]
@tbazant @LenzGr This PR is now updated with all the changes needed to replace `lrbd` with `ceph-iscsi`, so it's ready for a final review.

**Comment by ricardoasmarques** [comment]
CC'ing @ddiss - Just in case you have the chance to take a look into this PR (I've dropped a lot from the `tcmu-runner` section).

**Comment by ricardoasmarques** [comment]
@ddiss I found 3 advanced settings that are not documented but I don't have a small description for those settings, can you help me finding a description for those settings:
- force_pr_aptpl
- pi_prot_verify
- unmap_zeroes_data

**Comment by ddiss** [comment]
> CC'ing @ddiss - Just in case you have a chance to take a look into this PR (I've dropped a lot from the `tcmu-runner` section).

Thanks for the ping, putting this on my queue.

File: xml/deployment_iscsi.xml [hunk 3]
```diff
@@ -305,326 +305,159 @@
...
+> /iscsi-targets> create iqn.2003-01.org.linux-iscsi.iscsi.x86:testvol
+</screen>
    <para>
-    The &ceiscsi; "service" does not operate any background
-    daemon. Instead, it simply invokes the <command>lrbd</command> command.
-    This type of service is known as a "one-shot" service.
+    Create the iSCSI gateways by specifying the gateway <literal>name</literal>
+    and <literal>ip</literal> address:
    </para>
+<screen>
```
[the remark this answers is not in the document]
Comment (ricardoasmarques) [reply xml/deployment_iscsi.xml]: Yes, and it's how `gwcli` works:

Screenshot from 2019-04-24 09-10-19

**Comment by ricardoasmarques** [comment]
@tbazant I've addresses all your comments. Thanks.

**Pull request merged**